app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -244,8 +242,25 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
           if (previousFragment != null) {
             fragment.childFragmentManager.beginTransaction().remove(previousFragment).commitNow()
           }
+          val exitProfileDialogArguments =
+            if (getFooterViewModel().isAdministratorControlsSelected.get() == true) {
```

Comment:
This is the part I'm still confused about. Why do we need to specialize the behavior for admin controls/why does it need to be treated differently than everything else? I think we can probably avoid this tight coupling, but I need to better understand why admin controls is special.

Reply:
This is really a good point, so I am adding what I understand 

1. If we navigate between the fragments/activities associated with the NavDrawer Menu Items (Home, Options, Switch Profile, Help, Downloads), then the `NavigationDrawerMenuItem` itself manages the highlighting/unhighlighting the corresponding menuItem (we just need to set `.isChecked = true` for one item and it will automatically unhighlight the previous item).
2. But as `Administrator Controls` item is not a part of the NavigationDrawer menu item, we can't apply `.ischecked` property on it (so we explicitly change the color of text "Administrator Controls" when `isAdministratorControlsSelected` is true) so it doesn't automatically unhighlight the color of the previous menu item. Now, suppose we are in Administrator Controls Frgament, we open the NavDrawer, click on `Switch Profile`, then both of the items `Switch Profile` and `Administrator Controls` will get highlighted (Note that this isn't observed for the other menu items).
3. So, I am explicitly setting `isAdministratorControlsSelected` to false when we click `Switch Profile` and its restoration after canceling the dialog also requires separate implementation.

Reply:
Thanks for clarifying it. Given your explanation, I suspect this is probably the cleanest approach (at least I can't think of any way to further improve it). Thanks for the very thorough explanation!